### PR TITLE
Added exact package versions usage

### PR DESCRIPTION
**Pull request by oleksandr-danylchenko**
Fixes the wildcard bug that doesn't allow to build the project, see https://github.com/annotorious/annotorious/pull/386#issuecomment-2014975694. Specifically added usage of the `.19` version that doesn't have the "*"

### CODE REVIEW

**Comment by oleksandr-danylchenko** [comment]
Yep, the `build` passed successfully now: 
image

**Comment by oleksandr-danylchenko** [comment]
Oh, but using the `.19` version isn't be best option either, because it rolls back the pixi dependency in lock 🤔

**Comment by oleksandr-danylchenko** [comment]
Oh...

**Comment by oleksandr-danylchenko** [comment]
I think a better alternative would be to bump the versions across the packages and specify the new version everywhere instead of `.19` to avoid the unexpected rollback

**Comment by rsimon** [comment]
Yes, that's exactly what I thought was needed. But, ideally, in a script supported way. (Not that I'm lazy or anything... but I know already I'll keep forgetting to update in some places every time a roll a new release :-D) For now, I'd say let's just fix it manually. But in the long run, some kind of automated solution would be great. I'm sure there must be something out there. Just hadn't had the time yet to do a thorough search.

**Comment by oleksandr-danylchenko** [comment]
I suppose we can try using something like the [`bumpp`](https://www.npmjs.com/package/bumpp?activeTab=readme) that updates all the packages recursively. I'll try it out 👌🏻

**Comment by oleksandr-danylchenko** [comment]
> I suppose we can try using something like the [`bumpp`](https://www.npmjs.com/package/bumpp?activeTab=readme) that updates all the packages recursively. I'll try it out 👌🏻

Nah, it doesn't update the `dependencies` out of the box ;(

**Comment by oleksandr-danylchenko** [comment]
Seems like such a feature is supported only by `yarn` and its `yarn upgrade-interactive` command 🤷🏻‍♂️

**Comment by oleksandr-danylchenko** [comment]
@rsimon, what is the usual deploy pattern here? Do you start from `a9s/core` and go up through the dependencies list?
1. @annotorious/core
2. @annotorious/annotorious
3. @annotorious/openseadragon
4. @annotorious/react
5. @annotorious/react-manifold
6. @annotorious/svelte

**Comment by oleksandr-danylchenko** [comment]
Could I please get the `3.0.0-rc.21` deployed? 🙏🏻 🙏🏻🙏🏻
I'll unblock me with migration to the `@annotorious` dependencies in my project instead of littering NPM with "proxy" packages

**Comment by rsimon** [comment]
> Could I please get the 3.0.0-rc.21 deployed? 🙏🏻 🙏🏻🙏🏻
I'll unblock me with migration to the @annotorious dependencies in my project instead of littering NPM with "proxy" packages

The current latest state of the repo is published on npm as 3.0.0-rc.20 (from just an hour ago or so)

**Comment by oleksandr-danylchenko** [comment]
Yeah, but `.20` uses the `"*"` dependencies internally **_that don't allow building the packages_**. So the new version, where all the dependencies will point to the `^3.0.0-rc.21`, is needed

**Comment by oleksandr-danylchenko** [comment]
Also having the new version will unblock this PR - https://github.com/recogito/text-annotator-js/pull/75

File: packages/annotorious/package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "vitest": "^1.4.0"
   },
   "dependencies": {
-    "@annotorious/core": "*",
+    "@annotorious/core": "3.0.0-rc.19",
```
Comment (oleksandr-danylchenko) [inline packages/annotorious/package.json:51]: *Can only be bumped to the `^3.0.0-rc.21` after the respective version will get published on NPM

**Comment by oleksandr-danylchenko** [comment]
Please 🙏🏻 
It's the only think that blocks me from moving forward through my workday

**Comment by rsimon** [comment]
Ok, just to confirm. That would mean whenever a new version gets published (say, 3.0.0-rc.21) the dependencies could only ever point to the previous version, right? Hm, not ideal. But a) there's probably no way around that; b) when people use the package, a `^3.0.0-rc.20` would probably cause npm to pull the latest version (3.0.0-rc.21) anyway.

Re build order: I'm using npm workspaces, which defines it [in the monorepo package.json](https://github.com/annotorious/annotorious/blob/main/package.json). The order is as you say:

* annotorious-core
* annotorious
* annotorious-openseadragon
* annotorious-react
* annotorious-react-manifold
* annotorious-svelte

**Comment by rsimon** [comment]
I can publish the new release now. But I'll bump the dependency versions to ^3.0.0-rc.20, ok? I.e. I'll close this PR & make that change.

**Comment by oleksandr-danylchenko** [comment]
> That would mean whenever a new version gets published (say, 3.0.0-rc.21) the dependencies could only ever point to the previous version, right

Not exactly... The process to make them all use the same versions can look like this:
1. Bump all the versions to `.21`
2. Publish `@a9s/core` to NPM
3. Install `@a9s/core@.21` instead of `.19` in the packages
4. Repeat steps 2-3 in the workspaces order

In that way, all the packages will have the `.21` version and will depend on `.21` versions too

**Comment by oleksandr-danylchenko** [comment]
> b) when people use the package, a ^3.0.0-rc.20 would probably cause npm to pull the latest version (3.0.0-rc.21) anyway.

Also correct. Unless their package-lock has captured the `.20` version

**Comment by rsimon** [comment]
Yes, but that means you'd have to publish each package individually, and then, after publishing one package, update the version numbers in the next, publish the next, update version number in the next etc. Again, possible to script, but not my favourite workflow I guess.

**Comment by oleksandr-danylchenko** [comment]
> Yes, but that means you'd have to publish each package individually, and then, after publishing one package, update the version numbers in the next, publish the next,

Unfortunately, having the current setup, it's the only way to make everything use the same version 🤷🏻‍♂️

**Comment by rsimon** [comment]
v3.0.0-rc.21 is up!

**Comment by oleksandr-danylchenko** [comment]
Thanks!

**Pull request closed without merging**